lib/jekyll/site.rb

```diff
@@ -68,6 +68,7 @@ def reset
       self.static_files = []
       self.data = {}
       @collections = nil
+      @regenerator.clear_cache()
```

Comment:
I don't think we want to clear the metadata... it should still be a correct representation of what was generated previously. It's just the cache of whether or not a path needs to be regenerated that's now stale.